services/core-web/src/components/modalContent/RejectApplicationModal.js

```diff
@@ -1,33 +1,100 @@
-import React from "react";
+import React, { Component } from "react";
 import PropTypes from "prop-types";
-import { Alert } from "antd";
+import { Steps, Alert } from "antd";
+import CustomPropTypes from "@/customPropTypes";
+import GenerateDocumentForm from "@/components/Forms/GenerateDocumentForm";
 import RejectApplicationForm from "@/components/Forms/noticeOfWork/RejectApplicationForm";
 
 const propTypes = {
   onSubmit: PropTypes.func.isRequired,
   closeModal: PropTypes.func.isRequired,
   title: PropTypes.string.isRequired,
+  generateDocument: PropTypes.func.isRequired,
+  documentType: PropTypes.objectOf(PropTypes.any).isRequired,
+  type: PropTypes.string.isRequired,
+  signature: PropTypes.bool.isRequired,
+  draftAmendment: CustomPropTypes.permit.isRequired,
 };
 
-export const RejectApplicationModal = (props) => {
-  return (
-    <div>
-      <Alert
-        message="This action is final"
-        description="No changes or additions can be made to this application after the permit has been rejected."
-        type="warning"
-        showIcon
-        style={{ textAlign: "left" }}
-      />
-      <br />
-      <RejectApplicationForm
-        onSubmit={props.onSubmit}
-        closeModal={props.closeModal}
-        title={props.title}
-      />
-    </div>
-  );
-};
+export class RejectApplicationModal extends Component {
+  state = { step: 0, submitting: false };
+
+  handleGenerate = (values) => {
+    this.setState({ submitting: true });
+    this.props.generateDocument(this.props.documentType, values);
+    this.next();
+  };
+
+  close = () => {
+    this.props.closeModal();
+  };
+
+  next = () => this.setState((prevState) => ({ step: prevState.step + 1 }));
+
+  prev = () => this.setState((prevState) => ({ step: prevState.step - 1 }));
+
+  render() {
+    const steps = [
+      {
+        title: "Generate Letter",
+        content: (
+          <GenerateDocumentForm
+            {...this.props}
+            showActions={false}
+            additionalTitle="and Process"
+            onSubmit={this.handleGenerate}
+            submitting={this.state.submitting}
+            disabled={!this.props.signature}
+          />
+        ),
+      },
+      {
+        title: "Process",
+        content: (
+          <RejectApplicationForm
+            onSubmit={this.props.onSubmit}
+            closeModal={this.props.closeModal}
+            title={this.props.title}
+            type={this.props.type}
+            prev={this.prev}
+            draftAmendment={this.props.draftAmendment}
+          />
+        ),
+      },
+    ];
+    return (
+      <div>
+        <Alert
+          message="This action is final"
+          description="No changes or additions can be made to this application after it has been rejected."
```

Comment:
Let's say "after it has been **withdrawn**." if it is withdrawn and not rejected?

Reply:
The end status is always rejected - the difference between withdrawn is the different letter